gems/aws-sdk-cognitoidentity/lib/aws-sdk-cognitoidentity/customizations/cognito_identity_credentials.rb

```diff
@@ -0,0 +1,102 @@
+require 'set'
+require 'securerandom'
+require 'base64'
+
+module Aws
+  module CognitoIdentity
+    # An auto-refreshing credential provider
+    # TODO: Add good documentation....
+    class CognitoIdentityCredentials
+
+      include CredentialProvider
+      include RefreshingCredentials
+
+      # @option options [String] :identity_id the Cognito identity_id.  Required
```

Comment:
For Yard, I think you need `#param [Hash] options` or something similar above this.